src/main/java/de/rwth/idsg/steve/service/OcppTagService.java

```diff
@@ -97,35 +98,36 @@ public void removeUnknown(List<String> idTagList) {
     }
 
     @Nullable
-    public IdTagInfo getIdTagInfo(@Nullable String idTag, boolean isStartTransactionReqContext) {
+    public IdTagInfo getIdTagInfo(@Nullable String idTag, boolean isStartTransactionReqContext,
+        @Nullable String chargeBoxId, @Nullable Integer connectorId) {
         if (Strings.isNullOrEmpty(idTag)) {
             return null;
         }
 
-        OcppTagActivityRecord record = ocppTagRepository.getRecord(idTag);
-        AuthorizationStatus status = decideStatus(record, idTag, isStartTransactionReqContext);
-
-        switch (status) {
+        IdTagInfo idTagInfo = authTagService.decideStatus(idTag,
+            isStartTransactionReqContext, chargeBoxId, connectorId);
+        switch (idTagInfo.getStatus()) {
             case INVALID:
                 invalidOcppTagService.processNewUnidentified(idTag);
-                return new IdTagInfo().withStatus(status);
+                return idTagInfo;
 
             case BLOCKED:
             case EXPIRED:
             case CONCURRENT_TX:
             case ACCEPTED:
-                return new IdTagInfo().withStatus(status)
-                                      .withParentIdTag(record.getParentIdTag())
-                                      .withExpiryDate(getExpiryDateOrDefault(record));
+                return idTagInfo;
```

Comment:
we might question whether we need all the switch cases here (i think not). i guess we can just can use

```
if (idTagInfo.getStatus() == INVALID) {
    invalidOcppTagService.processNewUnidentified(idTag);
}

return idTagInfo
```